my-vault/internal/storage/queries/secret.sql

```diff
@@ -0,0 +1,12 @@
+
+-- name: PutNewSecret :exec
+INSERT INTO secrets (id, name, kv_data) VALUES ($1, $2, $3);
```

Comment:
I think we should omit the `ID` and let the database generate it for us. Also this could just be called `CreateSecret`. What do you think?
```suggestion
-- name: CreateSecret :exec
INSERT INTO secrets (name, kv_data) VALUES ($1, $2);
```

Also if you want to return the new secret (to show the ID to the user, for example), use `RETURNING`:

```sql
-- name: CreateSecret :one
INSERT INTO secrets (name, kv_data) VALUES ($1, $2) RETURNING *;
```

We also have to change the type to `:one` since we are reading a result row from the DB.